### PR TITLE
Add bulkDocs types

### PR DESCRIPTION
**Pull request by nohira**
case 1. Add a new type definition.
- [x] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [x] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [x] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

case 2. Improvement to existing type definition.
- documentation or source code reference which provides context for the suggested changes.  url http://api.jquery.com/html .
  - it has been reviewed by a DefinitelyTyped member.

Hi, I've added bulkDocs to db. I need it and I hope it helps a little.

### CODE REVIEW

File: pouchdb-core/pouchdb-core.d.ts [hunk 1]
```diff
@@ -268,6 +268,12 @@ declare namespace PouchDB {
         allDocs(options?: Core.AllDocsOptions):
             Promise<Core.AllDocsResponse<Content>>;
 
+        bulkDocs(docs: Core.Document<Content>[],
+                 options: Core.PutOptions | void,
```
Comment (geppy) [inline pouchdb-core/pouchdb-core.d.ts]: It looks like `bulkDocs` takes an `Array<Core.PutOptions>` (compare [`put` documentation](https://pouchdb.com/api.html#create_document) to [`bulkDocs` documentation](https://pouchdb.com/api.html#batch_create)).

File: pouchdb-core/pouchdb-core.d.ts [hunk 2]
```diff
@@ -268,6 +268,12 @@ declare namespace PouchDB {
...
+                 callback: Core.Callback<Core.Error, Core.Response[]>): void;
+        bulkDocs(docs: Core.Document<Content>[],
+                 options?: Core.PutOptions): Promise<Core.Response[]>;
```
Comment (geppy) [inline pouchdb-core/pouchdb-core.d.ts]: Same as above, I think this should be `Array<Core.PutOptions>`.

**Review by geppy — CHANGES_REQUESTED** [review]
Thanks!  This looks almost perfect, but there's one issue I noticed.

File: pouchdb-core/pouchdb-core.d.ts (hunk 2 above)
Comment (nohira) [inline pouchdb-core/pouchdb-core.d.ts]: Thanks. Actually, I can't find where it's said that it's an array.
I was on pouchdb slack and I had the confirmation from someone that it's just an object.
In addition, this is some go implementation:
[Go implementation of pouchdb](https://godoc.org/github.com/flimzy/go-pouchdb#PouchDB.BulkDocs)

Comment (geppy) [inline pouchdb-core/pouchdb-core.d.ts]: Oh, you're absolutely right.  I misread your type definitions (I was looking at the wrong parameter).   Looks good to me.

**Review by geppy — APPROVED** [review]

**Pull request merged**